spec/helpers.lua

```diff
@@ -1697,13 +1697,41 @@ local function render_fixtures(conf, env, prefix, fixtures)
   return true
 end
 
+
+local function build_go_plugins(path)
+  for _, plugin_path in ipairs(pl_dir.getfiles(path, "*.go")) do
+    local plugin_name = pl_path.basename(plugin_path):match("(.+).go")
+
+    local ok, _, _, stderr = pl_utils.executeex(
+      string.format("go build -buildmode plugin -o %s %s",
+      path .. "/" .. plugin_name .. ".so", plugin_path)
+    )
+    assert(ok, stderr)
+  end
+end
+
 local function start_kong(env, tables, preserve_prefix, fixtures)
   if tables ~= nil and type(tables) ~= "table" then
     error("arg #2 must be a list of tables to truncate")
   end
   env = env or {}
   local prefix = env.prefix or conf.prefix
 
+  -- go plugins are enabled
+  --  set pluginserver dir (making sure it's in the PATH)
+  --  compile fixture go plugins
+  if env.go_plugins_dir then
+    if env.go_plugins_dir == GO_PLUGIN_PATH then
+      build_go_plugins(GO_PLUGIN_PATH)
+    end
+
+    if not env.go_pluginserver_exe and not os.getenv("KONG_GO_PLUGINSERVER_EXE") then
```

Comment:
@javierguerragiraldez This is what I meant - only looking up in PATH if none of these have been set, as it'd otherwise potentially interfere with the use of KONG_GO_PLUGINSERVER_EXE in development workflows.

Reply:
Sorry, I commented over an outdated commit earlier.

I still think this is not really _necessary_, but since the environment variable is taking precedence in this version, I think it doesn't hurt either. So, :+1: